frontend/src/metabase/components/tree/TreeNode.tsx

```diff
@@ -11,67 +11,60 @@ import {
   NameContainer,
   IconContainer,
 } from "./TreeNode.styled";
-import { ITreeNodeItem } from "./types";
-
-export interface TreeNodeProps {
-  item: ITreeNodeItem;
-  depth: number;
-  hasChildren: boolean;
-  isExpanded: boolean;
-  isSelected: boolean;
-  colorScheme: "default" | "admin";
-  onSelect: (item: ITreeNodeItem) => void;
-  onToggleExpand: (id: ITreeNodeItem["id"]) => void;
-}
+import { TreeNodeProps } from "./types";
 
 // eslint-disable-next-line react/display-name
-export const TreeNode = React.memo(
+const BaseTreeNode = React.memo(
   React.forwardRef<HTMLLIElement, TreeNodeProps>(function TreeNode(
     {
+      item,
+      depth,
       isExpanded,
       isSelected,
       hasChildren,
-      onToggleExpand,
       onSelect,
-      depth,
-      item,
-      colorScheme,
+      onToggleExpand,
+      ...props
     },
     ref,
   ) {
-    const { name, icon, id } = item;
+    const { name, icon } = item;
 
     const iconProps = _.isObject(icon) ? icon : { name: icon };
 
-    const handleSelect = () => {
-      onSelect(item);
-      onToggleExpand(id);
-    };
+    function onClick() {
+      if (typeof onSelect === "function") {
+        onSelect();
+      }
+      onToggleExpand();
+    }
 
     const handleKeyDown: React.KeyboardEventHandler = ({ key }) => {
       switch (key) {
         case "Enter":
-          onSelect(item);
+          if (typeof onSelect === "function") {
+            onSelect();
```

Comment:
nit: `onSelect?.();`

Reply:
Thanks, I can't memorize this syntax at all